typing_extensions/src_py3/typing_extensions.py

```diff
@@ -620,3 +664,300 @@ def new_type(x):
+        # includes "Protocol" special treatment. (Comments removed for brevity.)
+        assert extra is None  # Protocols should not have extra
+        if tvars is not None:
+            assert origin is not None
+            assert all(isinstance(t, TypeVar) for t in tvars), tvars
+        else:
+            tvars = _type_vars(bases)
+            gvars = None
+            for base in bases:
+                if base is Generic:
+                    raise TypeError("Cannot inherit from plain Generic")
+                if (isinstance(base, GenericMeta) and
+                        base.__origin__ in (Generic, Protocol)):
+                    if gvars is not None:
+                        raise TypeError(
+                            "Cannot inherit from Generic[...] or"
+                            " Protocol[...] multiple times.")
+                    gvars = base.__parameters__
+            if gvars is None:
+                gvars = tvars
+            else:
+                tvarset = set(tvars)
+                gvarset = set(gvars)
+                if not tvarset <= gvarset:
+                    raise TypeError(
+                        "Some type variables (%s) "
+                        "are not listed in %s[%s]" %
+                        (", ".join(str(t) for t in tvars if t not in gvarset),
+                         "Generic" if any(b.__origin__ is Generic
+                                          for b in bases) else "Protocol",
+                         ", ".join(str(g) for g in gvars)))
+                tvars = gvars
+
+        initial_bases = bases
+        if extra is not None and type(extra) is abc.ABCMeta and extra not in bases:
+            bases = (extra,) + bases
+        bases = tuple(_gorg(b) if isinstance(b, GenericMeta) else b for b in bases)
+        if any(isinstance(b, GenericMeta) and b is not Generic for b in bases):
+            bases = tuple(b for b in bases if b is not Generic)
+        namespace.update({'__origin__': origin, '__extra__': extra})
+        self = super(GenericMeta, cls).__new__(cls, name, bases, namespace, _root=True)
+        super(GenericMeta, self).__setattr__('_gorg',
+                                             self if not origin else _gorg(origin))
+        self.__parameters__ = tvars
+        self.__args__ = tuple(... if a is _TypingEllipsis else
+                              () if a is _TypingEmpty else
+                              a for a in args) if args else None
+        self.__next_in_mro__ = _next_in_mro(self)
+        if orig_bases is None:
+            self.__orig_bases__ = initial_bases
+        elif origin is not None:
+            self._abc_registry = origin._abc_registry
+            self._abc_cache = origin._abc_cache
+        if hasattr(self, '_subs_tree'):
+            self.__tree_hash__ = (hash(self._subs_tree()) if origin else
+                                  super(GenericMeta, self).__hash__())
+        return self
+    if OLD_GENERICS:
```

Comment:
Why not just put the `def` inside `if not OLD_GENERICS:`? Just so you can copy/paste without reindenting? That feels a little weird to me.

Reply:
I just wanted to save some horizontal space. But I see that it is easy to miss these two lines when scrolling through the code. I will probably just put all definitions inside (nested) ``if`` statements.